### DISCUSSION TITLE
Double Vlan tagging on the FH packets

### DISCUSSION THREAD

**Discussion by ajesh-pk** [srsran/srsRAN_Project#740] [General Help] [answered]
Somehow I get double Vlan tagged by the dpdk application, because of which the RU is marking it as bad packets. I captured  the packets form switch, can you suggest how to solve this.

`
ran_node_name: gnb

amf:
  n2_bind_addr: 10.2.2.54                                              # The address or hostname of the AMF.
  n3_bind_addr: 10.2.3.54                                        # A local IP that the gNB binds to for traffic from the AMF.
  addr: 10.2.2.177

hal:
        eal_args: "--lcores (1,3)@(1,3,5,7,9,11,13,15,17,19,21,23,25,27,29) -a 0000:b1:01.0"

ru_ofh:
  t1a_max_cp_dl: 470                                              # Maximum T1a on Control-Plane for Downlink in microseconds.
  t1a_min_cp_dl: 419                                              # Minimum T1a on Control-Plane for Downlink in microseconds.
  t1a_max_cp_ul: 336                                              # Maximum T1a on Control-Plane for Uplink in microseconds.
  t1a_min_cp_ul: 285                                              # Minimum T1a on Control-Plane for Uplink in microseconds.
  t1a_max_up: 345                                                 # Maximum T1a on User-Plane in microseconds.
  t1a_min_up: 294                                                 # Minimum T1a on User-Plane in microseconds.
  ta4_max: 331                                                    # Maximum Ta4 on User-Plane in microseconds.
  ta4_min: 50                                                     # Minimum Ta4 on User-Plane in microseconds.
  is_prach_cp_enabled: true                                       # Configures if Control-Plane messages should be used to receive PRACH messages.
  compr_method_ul: bfp                                            # Uplink compression method.
  compr_bitwidth_ul: 9                                            # Uplink IQ samples bitwidth after compression.
  compr_method_dl: bfp                                            # Downlink compression method.
  compr_bitwidth_dl: 9                                            # Downlink IQ samples bitwidth after compression.
  compr_method_prach: bfp                                         # PRACH compression method.
  compr_bitwidth_prach: 9                                         # PRACH IQ samples bitwidth after compression.
  enable_ul_static_compr_hdr: true                                # Configures if the compression header is present for uplink User-Plane messages (false) or not present (true).
  enable_dl_static_compr_hdr: true                                # Configures if the compression header is present for downlink User-Plane messages (false) or not present (true).
  iq_scaling: 9                                                 # IQ samples scaling factor applied before compression, should be a positive value smaller than 10.
  cells:                              
    - network_interface: 0000:b1:01.0                               # Ethernet interface name used to communicate with the RU.
      ru_mac_addr: 98:ae:71:04:83:bb                                # RU MAC address.
      du_mac_addr: 00:11:22:33:44:66                                # DU MAC address.
      vlan_tag_cp: 2                                                # VLAN tag value for C-Plane.
      vlan_tag_up: 2                                                # VLAN tag value for U-Plane.
      prach_port_id: [4, 5, 6, 7] # PRACH eAxC port value.
      dl_port_id: [0, 1, 2, 3] # Downlink eAxC port values.
      ul_port_id: [0, 1, 2, 3] # Uplink eAxC port values.

cell_cfg:
  dl_arfcn: 649884                                                 # ARFCN of the downlink carrier (center frequency).
  band: 78                                                        # The NR band.
  channel_bandwidth_MHz: 100                                      # Bandwith in MHz. Number of PRBs will be automatically derived.
  common_scs: 30                                                  # Subcarrier spacing in kHz used for data.
  plmn: "090170"                                                   # PLMN broadcasted by the gNB.
  tac: 1                                                          # Tracking area code (needs to match the core configuration).
  pci: 1                                                          # Physical cell ID.
  nof_antennas_dl: 4                                              # Downlink number of antennas.
  nof_antennas_ul: 4                                              # Uplink number of anntennas.
  prach:
    prach_config_index: 159                                         # PRACH configuration index.
    prach_root_sequence_index: 0                                    # PRACH root sequence index.
    zero_correlation_zone: 0                                        # Zero correlation zone.
    prach_frequency_start: 0                                        # Offset in PRBs of lowest PRACH transmission occasion in frequency domain respective to PRB 0.
  pucch:
          #sr_period_ms: 20   			                            # Sets the SR period in milliseconds. This can be set either 20 or 40 ms.
   f1_nof_cell_res_sr: 15                                          # Sets the number of PUCCH F1 resources available per cell for SR.
   f2_nof_cell_res_csi: 15                                         # Sets the number of PUCCH F2 resources available per cell for CSI. 
  csi:
          #csi_rs_period: 40  					            # Sets the CSI-RS period in milliseconds. This can be set either 20 or 40 ms.
  pusch:
    dmrs_additional_position: 2                                     # Sets the PUSCH DMRS additional position.
  tdd_ul_dl_cfg:
    dl_ul_tx_period: 5
    nof_dl_symbols: 10                                               # Number of Downlink symbols at the beginning of the slot following full Downlink slots.
    nof_ul_symbols: 2                                               # Number of Uplink symbols at the end of the slot preceding the first full Uplink slot.
    nof_dl_slots: 3                                                 # Number of consecutive full Downlink slots.
    nof_ul_slots: 1                                                 # Number of consecutive full Uplink slots.

log:
  filename: /tmp/gnb.log                                          # Path of the log file.
  all_level: info                                                 # Logging level applied to all layers.

pcap:
  mac_enable: false                                               # Set to true to enable MAC-layer PCAPs.
  mac_filename: /tmp/gnb_mac.pcap                                 # Path where the MAC PCAP is stored.
  ngap_enable: false                                              # Set to true to enable NGAP PCAPs.
  ngap_filename: /tmp/gnb_ngap.pcap                               # Path where the NGAP PCAP is stored.
`

[SRS (1).zip](https://github.com/user-attachments/files/16309308/SRS.1.zip)

**Comment by andrepuschmann** [comment, marked as answer]
Hey @ajesh-pk - please disable VLAN tagging on your virtual function as our OFH implementation will add the VLANs itself.
